File: 2/2000s/22s/2200/FindAllK-DistantIndicesInAnArray.java
/*
https://leetcode.com/problems/find-all-k-distant-indices-in-an-array/description/?envType=daily-question&envId=2025-06-24
*/

class Solution {
    public List<Integer> findKDistantIndices(int[] nums, int key, int k) {
        ArrayList<Integer>resultantKDistantIndices=new ArrayList<>();
        ArrayList<Integer>indicesWithKey=new ArrayList<>();
        int lengthOfNums=nums.length;
        for(int i=0; i<lengthOfNums; ++i){
            if(nums[i]==key){
                indicesWithKey.add(i);
            }
        }
        for(int i=0; i<lengthOfNums; ++i){
            int currentIndex=i;
            int sizeOfIndicesLst=indicesWithKey.size();
            for(int j=0; j<sizeOfIndicesLst; ++j){
                int nxtIndex=indicesWithKey.get(j);
                if(Math.abs(currentIndex-nxtIndex)<=k){
                    resultantKDistantIndices.add(i);
                    break;
                }
            }
        }
        return resultantKDistantIndices;
    }
}

File: 2/2000s/22s/2200/FindAllK-DistantIndicesInAnArray.cpp
/*
https://leetcode.com/problems/find-all-k-distant-indices-in-an-array/description/?envType=daily-question&envId=2025-06-24
*/

class Solution {
public:
    vector<int> findKDistantIndices(vector<int>& nums, int key, int k) {
        vector<int>res, kyIndices;
        int sz=nums.size();
        for(int i=0; i<sz; ++i){
            if(nums[i]==key){
                kyIndices.push_back(i);
            }
        }
        for(int i=0; i<sz; ++i){
            int curidx=i;
            for(auto &vals: kyIndices){
                if(abs(curidx-vals)<=k){
                    res.push_back(i);
                    break;
                }
            }
        }
        return res;
    }
};